### PR TITLE
 Add form question mutations 

### PR DESCRIPTION
**Pull request by sliverc**
Allow add, remove and reorder

### CODE REVIEW

File: caluma/form/schema.py [hunk 1]
```diff
@@ -41,6 +48,90 @@ def mutate_and_get_payload(cls, root, info, **input):
...
+        question_ids = graphene.List(graphene.ID, required=True)
+
+    form = graphene.Field(Form)
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **input):
+        _, form_id = from_global_id(input["form_id"])
+        form = get_object_or_404(models.Form, pk=form_id)
+
+        curr_questions = form.questions.values_list("slug", flat=True)
```
[the remark this answers is not in the document]
Comment (sliverc) [reply caluma/form/schema.py]: we allow reordering of a form once it is published as this is a trivial change

File: caluma/form/schema.py [hunk 2]
```diff
@@ -41,6 +48,90 @@ def mutate_and_get_payload(cls, root, info, **input):
...
+        _, form_id = from_global_id(input["form_id"])
+        form = get_object_or_404(models.Form, pk=form_id)
+
+        if form.is_archived or form.is_published:
+            raise GraphQLError(
+                "Form %s may not be edited as it is archived or published" % form_id
+            )
+
+        _, question_id = from_global_id(input["question_id"])
+        question = get_object_or_404(models.Question, pk=question_id)
```
[the remark this answers is not in the document]
Comment (sliverc) [reply caluma/form/schema.py]: Fair point. Unfortunately graphene currently does support an abstraction layer for validation so meaning all needs to be handmade.

For now I have extracted the logic and moved it to a model method with a TODO that this needs to be rethought in the context of graphene.

**Comment by sliverc** [comment]
@winged Does it look ok for you now?

**Comment by winged** [comment]
Yep looks good now, thanks!
Regarding slug usage - this was explained by @anehx, so OK for me too

**Pull request merged**